### PR TITLE
redefine eye, and add Cartesian indexing

### LINKED ISSUE #308: Add method `eye` for `SparseMatrixExp`
something like:

```
Base.eye(spmexp::SparseMatrixExp) = eye(spmexp.M)
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #308.

### CODE REVIEW

File: src/ExponentialMap.jl [hunk 1]
```diff
@@ -61,7 +61,10 @@ end
 SparseMatrixExp(M::Matrix{N}) where {N} =
         error("only sparse matrices can be used to create a `SparseMatrixExp`")
 
-Base.eye(spmexp::SparseMatrixExp) = eye(spmexp.M)
+Base.eye(spmexp::SparseMatrixExp) = SparseMatrixExp(zeros(spmexp.M))
```
Comment (schillic) [inline src/ExponentialMap.jl]: `spzeros` would be more efficient, right?

File: src/ExponentialMap.jl [hunk 2]
```diff
@@ -61,7 +61,10 @@ end
...
+
+Base.IndexStyle(::Type{<:SparseMatrixExp}) = IndexCartesian()
+Base.getindex(spmexp::SparseMatrixExp{N}, I::Vararg{Int, 2}) where {N} = get_column(spmexp, I[2])[I[1]]
```
Comment (schillic) [inline src/ExponentialMap.jl]: I think you can omit the type `N` if you do not use it on the r.h.s.

File: src/ExponentialMap.jl (hunk 1 above)
Comment (schillic) [reply src/ExponentialMap.jl]: Yes, that is what I meant.

File: src/ExponentialMap.jl [hunk 3]
```diff
@@ -61,10 +61,10 @@ end
 SparseMatrixExp(M::Matrix{N}) where {N} =
         error("only sparse matrices can be used to create a `SparseMatrixExp`")
 
-Base.eye(spmexp::SparseMatrixExp) = SparseMatrixExp(zeros(spmexp.M))
+Base.eye(spmexp::SparseMatrixExp) = SparseMatrixExp(spzeros(size(E.M)...))
```
Comment (schillic) [inline src/ExponentialMap.jl]: What is `E`?

**Comment by mforets** [comment]
merge?

**Pull request merged**